File: src/main/java/org/svomz/apps/finances/domain/AccountRepository.java
package org.svomz.apps.finances.domain;

import java.util.Optional;
import java.util.Set;

/**
 * Created by eric on 28/02/16.
 */
public interface AccountRepository {

  Account create(Account account);

  Set<Account> findAll();

  Optional<Account> find(String id);

  AccountId nextIdentity();
}

File: src/main/scala/org/svomz/apps/finances/adapter/secondary/persistence/AccountRepository.scala
package org.svomz.apps.finances.adapter.secondary.persistence

import java.util.UUID

import org.svomz.apps.finances.domain.model.{Account, AccountRepository}

import scala.collection.mutable.MutableList

object AccountRepository extends AccountRepository[String] {
  var accounts: MutableList[Account] = MutableList.empty

  override def persist(account: Account): Account = {
    accounts :+ account
    account
  }

  override def fetch(no: String): Option[Account] = accounts.filter(_.no.equals(no)).headOption
  override def nextAccountNumber: String = UUID.randomUUID().toString
}
